communications/models.py

```diff
@@ -1,3 +1,192 @@
 from django.db import models
+from django.conf import settings
+from django.utils.translation import gettext_lazy as _
 
-# Create your models here.
+
+class NotificationFrequency(models.TextChoices):
+    IMMEDIATE = 'immediate', _('Immediate')
+    DAILY_DIGEST = 'daily_digest', _('Daily Digest')
+    WEEKLY_SUMMARY = 'weekly_summary', _('Weekly Summary')
+    DISABLED = 'disabled', _('Disabled')
+
+
+class NotificationType(models.Model):
+    """Model to store different types of notifications."""
+    code = models.SlugField(
+        max_length=50,
+        unique=True,
+        help_text=_('Unique code for the notification type')
+    )
+    name = models.CharField(
+        max_length=100,
+        help_text=_('Human-readable name')
+    )
+    description = models.TextField(
+        blank=True,
+        help_text=_('Description of when this notification is sent')
+    )
+    default_frequency = models.CharField(
+        max_length=20,
+        choices=[
+            ('immediate', _('Immediate')),
+            ('daily_digest', _('Daily Digest')),
+            ('weekly_summary', _('Weekly Summary')),
+            ('disabled', _('Disabled'))
+        ],
+        default='immediate',
+        help_text=_('Default frequency for this notification type')
+    )
+    is_active = models.BooleanField(
+        default=True,
+        help_text=_('Whether this notification type is active')
+    )
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        ordering = ['name']
+        verbose_name = _('Notification Type')
+        verbose_name_plural = _('Notification Types')
+
+    def __str__(self):
+        return self.name
+
+    @classmethod
+    def get_default_pk(cls):
+        """Get or create a default notification type to use as a fallback.
+        
+        This is useful for database migrations and testing.
+        
+        Returns:
+            int: The primary key of the default notification type
+        """
+        obj, _ = cls.objects.get_or_create(
+            code='default',
+            defaults={
+                'name': 'Default Notification',
+                'description': 'Default notification type',
+            }
+        )
+        return obj.pk
+
+
+class Notification(models.Model):
+    """Model to store all notifications sent to users."""
+    notification_id = models.UUIDField(unique=True, editable=False, primary_key=True)
```

Comment:
Consider using `UUIDField` with `default=uuid.uuid4` to automatically generate UUIDs if your logic allows. Otherwise, ensure that `notification_id` is set at creation.